plugins/airtable/src/SelectDataSource.tsx

```diff
@@ -96,47 +116,58 @@ export function SelectDataSource({ collection, onSelectDataSource }: SelectDataS
-                    <option value="" disabled>
-                        {status === "loading-bases" ? "Loading…" : "Choose…"}
-                    </option>
-                    {bases.map(({ id, name }) => (
-                        <option key={id} value={id}>
-                            {name}
+            <div className="setup-list">
+                <label htmlFor="base">
+                    Base
+                    <select
+                        id="base"
+                        onChange={event => {
+                            setSelectedBaseId(event.target.value)
+                        }}
+                        value={selectedBaseId}
+                        disabled={status === "loading-bases" || status === "error-bases"}
+                    >
+                        <option value="" disabled>
+                            {status === "loading-bases" ? "Loading…" : status === "error-bases" ? "Error" : "Choose…"}
                         </option>
-                    ))}
-                </select>
-            </label>
-
-            <label htmlFor="table">
-                Table
-                <select
-                    id="table"
-                    onChange={event => {
-                        setSelectedTableId(event.target.value)
-                    }}
-                    value={selectedTableId}
-                    disabled={!selectedBaseId || status === "loading-tables"}
-                >
-                    <option value="" disabled>
-                        {status === "loading-tables" ? "Loading…" : "Choose…"}
-                    </option>
-                    {tables.map(({ id, name }) => (
-                        <option key={id} value={id}>
-                            {name}
+                        {bases.map(({ id, name }) => (
+                            <option key={id} value={id}>
+                                {name}
+                            </option>
+                        ))}
+                    </select>
+                </label>
+
+                <label htmlFor="table">
+                    Table
+                    <select
+                        id="table"
+                        onChange={event => {
+                            setSelectedTableId(event.target.value)
+                        }}
+                        value={selectedTableId}
+                        disabled={
+                            !selectedBaseId ||
+                            status === "loading-tables" ||
+                            status === "error-tables" ||
+                            status === "error-bases"
+                        }
+                    >
+                        <option value="" disabled>
+                            {status === "loading-tables"
+                                ? "Loading…"
+                                : status === "error-tables" || status === "error-bases"
+                                  ? "Error"
+                                  : "Choose…"}
```

Comment:
Try to avoid nested ternary operators because they can be difficult to follow and understand. Instead, consider wrapping decision tree into a helper function that uses a switch/case statement.